src/main/kotlin/org/rust/lang/core/resolve/RustResolveEngine.kt

```diff
@@ -66,8 +69,20 @@ object RustResolveEngine {
     fun resolveUseGlob(ref: RustUseGlob): ResolveResult =
         Resolver().resolveUseGlob(ref)
 
-}
+    fun resolveExternCrate(crate: RustExternCrateItem): ResolveResult {
```

Comment:
This does not delegate to `Resolver()` because there is no such need. `Resolver` holds resolve context, and we can resolve `extern crate` without any context. In fact, I think we with `resolveModDecl` (two functions up) we also should avoid this unnecessary delegation.